src/Centreon/Infrastructure/PlatformTopology/PlatformTopologyRepositoryRDB.php

```diff
@@ -0,0 +1,104 @@
+use Centreon\Infrastructure\DatabaseConnection;
+use Centreon\Infrastructure\Repository\AbstractRepositoryDRB;
+
+/**
+ * This class is designed to manage the repository of the platform topology requests
+ *
+ * @package Centreon\Infrastructure\PlatformTopology
+ */
+class PlatformTopologyRepositoryRDB extends AbstractRepositoryDRB implements PlatformTopologyRepositoryInterface
+{
+    /**
+     * PlatformTopologyRepositoryRDB constructor.
+     * @param DatabaseConnection $db
+     */
+    public function __construct(DatabaseConnection $db)
+    {
+        $this->db = $db;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function addPlatformToTopology(PlatformTopology $platformTopology): void
+    {
+        $request = $this->translateDbName('
+            INSERT INTO `:db`.platform_topology (`address`, `hostname`, `server_type`, `parent_id`)
+            VALUES (:serverAddress, :serverName, :serverType, :parentId)
+        ');
+
+        $statement = $this->db->prepare($request);
+        $statement->bindValue(':serverAddress', $platformTopology->getServerAddress(), \PDO::PARAM_STR);
+        $statement->bindValue(':serverName', $platformTopology->getServerName(), \PDO::PARAM_STR);
+        $statement->bindValue(':serverType', $platformTopology->getServerType(), \PDO::PARAM_INT);
+        $statement->bindValue(':parentId', $platformTopology->getServerParentId(), \PDO::PARAM_INT);
+        $statement->execute();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function findRegisteredPlatformsInTopology(string $serverAddress, string $serverName): array
+    {
+        $request = $this->translateDbName(
+            'SELECT `address`, `hostname`, `server_type`
+            FROM `:db`.platform_topology
+            WHERE `address` = :serverAddress OR `hostname` = :serverName'
+        );
+        $statement = $this->db->prepare($request);
+        $statement->bindValue(':serverAddress', $serverAddress, \PDO::PARAM_STR);
+        $statement->bindValue(':serverName', $serverName, \PDO::PARAM_STR);
+        $statement->execute();
+
+        $result = $statement->fetch(\PDO::FETCH_ASSOC);
+
+        // if nothing is found, convert $result to an empty array as expected
+        return (is_array($result) ? $result : []);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function findParentInTopology(string $serverAddress): array
+    {
+        $request = $this->translateDbName(
+            'SELECT platform_topology.id AS parent_id
+            FROM `:db`.platform_topology
+            WHERE address = :address'
+        );
+        $statement = $this->db->prepare($request);
+        $statement->bindValue(':address', $serverAddress, \PDO::PARAM_STR);
+        $statement->execute();
+
+        $result = $statement->fetch(\PDO::FETCH_ASSOC);
+
+        // if nothing is found, convert $result to an empty array as expected
+        return (is_array($result) ? $result : []);
```

Comment:
you should return entities

Reply:
Don't forget to update the interface afterwards.